server/src/main/java/io/druid/segment/realtime/RealtimeManager.java

```diff
@@ -325,6 +299,134 @@ public void run()
       }
     }
 
+    private void runFirehoseV2(FirehoseV2 firehose)
+    {
+      final Period intermediatePersistPeriod = config.getIntermediatePersistPeriod();
+      try {
+        firehose.start();
+      }
+      catch (Exception e) {
+      	log.error(e, "Failed to start firehoseV2");
+      	return;
+      }
+      long nextFlush = new DateTime().plus(intermediatePersistPeriod).getMillis();
+      log.info("Firehose started with nextFlush [%s]", nextFlush);
+      boolean haveRow = true;
+      while (haveRow) {
+        InputRow inputRow = null;
+        int numRows = 0;
+        try {
+          inputRow = firehose.currRow();
+          if (inputRow != null) {
+            try {
+              numRows = plumber.add(inputRow);
+            }
+            catch (IndexSizeExceededException e) {
+              log.info("Index limit exceeded: %s", e.getMessage());
+              nextFlush = doIncrementalPersist(firehose.makeCommitter(), intermediatePersistPeriod);
+              continue;
+            }
+
+            if (numRows < 0) {
+              metrics.incrementThrownAway();
+              log.debug("Throwing away event[%s]", inputRow);
+            } else {
+              metrics.incrementProcessed();
+            }
+          }
+        }
+        catch (Exception e) {
```

Comment:
While I don't disagree, we chose to skip the event because it's very possible that the event in question caused an exception due to something wrong with the event.  It is likely a bug, yes.  But in the community there have definitely been issues with exceptions that stop the real-time dead cold and then people are like, "what happened, my process died."

This code can change, it's not set in stone and it has to be "turned on" by someone actually using the new way of doing things.  I don't think this is worthy of blocking the PR.